aptos-move/move-examples/post_mint_reveal_nft/sources/whitelist.move

```diff
@@ -0,0 +1,102 @@
+module post_mint_reveal_nft::whitelist {
+    use post_mint_reveal_nft::bucket_table::BucketTable;
+    use std::vector;
+    use std::error;
+    use post_mint_reveal_nft::bucket_table;
+    use std::signer;
+    use aptos_framework::timestamp;
+    use aptos_framework::account;
+    friend post_mint_reveal_nft::minting;
+
+    /// WhitelistMintConfig stores information about all stages of whitelist.
+    /// Most whitelists are one-stage, but we allow multiple stages to be added in case there are multiple rounds of whitelists.
+    struct WhitelistMintConfig has key {
+        whitelist_configs: vector<WhitelistStage>,
+    }
+
+    /// WhitelistMintConfigSingleStage stores information about one stage of whitelist.
+    struct WhitelistStage has store {
+        whitelisted_address: BucketTable<address, u64>,
+        whitelist_mint_price: u64,
+        whitelist_minting_start_time: u64,
+        whitelist_minting_end_time: u64,
+    }
+
+    const EINVALID_WHITELIST_SETTING: u64 = 1;
+    const EINVALID_STAGE: u64 = 2;
+    const EACCOUNT_DOES_NOT_EXIST: u64 = 3;
+    const EEXCEEDS_MINT_LIMIT: u64 = 4;
+    const EINVALID_UPDATE_AFTER_MINTING: u64 = 5;
+
+
+    public(friend) fun whitelist_config_exists(admin: &signer): bool {
+        exists<WhitelistMintConfig>(signer::address_of(admin))
+    }
+
+    public(friend) fun is_whitelist_minting_time(module_address: address): bool acquires WhitelistMintConfig {
+        let whitelist_mint_config = borrow_global_mut<WhitelistMintConfig>(module_address);
+        let first_stage_start_time = vector::borrow(&whitelist_mint_config.whitelist_configs, 0).whitelist_minting_start_time;
+        let last_stage_end_time = vector::borrow(&whitelist_mint_config.whitelist_configs, vector::length(&whitelist_mint_config.whitelist_configs) - 1).whitelist_minting_end_time;
+        let now = timestamp::now_seconds();
+        now >= first_stage_start_time && now < last_stage_end_time
+    }
+
+    public(friend) fun get_num_of_stages(module_address: address): u64 acquires WhitelistMintConfig {
```

Comment:
Can you add comments for each public function explaining what it does?

Reply:
yup will do